core/utils/cuckoo_map_test.cc

```diff
@@ -45,6 +45,35 @@ TEST(CuckooMapTest, Insert) {
   EXPECT_EQ(cuckoo.Insert(1, 1)->second, 1);
 }
 
+struct Foo {
+  Foo() : a(), b(), c() {}
+
+  explicit Foo(int aa, int bb, int cc) : a(aa), b(bb), c(cc) {}
+
+  int a;
+  int b;
+  int c;
+};
+
+// Test Move-Insert function
+TEST(CuckooMapTest, MoveInsert) {
+  CuckooMap<uint32_t, Foo> cuckoo;
+  Foo expected = Foo(1, 2, 3);
+  const Foo &x = cuckoo.Insert(1, expected)->second;
```

Comment:
I think it ends up calling the copy one, when used without `std::move(expected)`

Reply:
That's what I try to understand and point out! :)